src/packages/@highlighter/control/NodeIterator/NodeIterator.ts

```diff
@@ -0,0 +1,46 @@
+interface NodeIteratorOptions {
+    /**
+     * Defines root node at which to begin the NodeIterator's traversal
+     */
+    root: HTMLElement;
+    /**
+     * Defines filtering for certain types of node
+     * https://developer.mozilla.org/en-US/docs/Web/API/Document/createNodeIterator
+     */
+    whatToShow?: number;
+}
+
+/**
+ * This module allows you to iterate through dom nodes
+ * Support filtering for certain types of node
+ */
+export class NodeIterator {
+    private root: HTMLElement;
+
+    private whatToShow: number | undefined;
```

Comment:
can be replaced with `private whatToShow?: number`?

Reply:
Good point!